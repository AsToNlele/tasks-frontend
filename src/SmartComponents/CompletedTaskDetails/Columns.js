import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { renderColumnComponent } from '../../Utilities/helpers';

const SystemNameCell = ({ /*id,*/ display_name }, index) => (
  <Link
    key={`system-title-${index}`}
    //to={`/executed/${id}`}
    to={`/`}
  >
    {display_name}
  </Link>
);

SystemNameCell.propTypes = {
  id: propTypes.string,
  display_name: propTypes.node,
};

export const SystemColumn = {
  title: 'System name',
  props: {
    width: 35,
  },
  sortByProp: 'display_name',
  renderExport: (job) => job.display_name,
  renderFunc: renderColumnComponent(SystemNameCell),
};

export const ResultColumn = {
  title: 'Result',
  props: {
    width: 20,
  },
  sortByProp: 'status',
  renderExport: (job) => job.status,
};

export const MessageColumn = {
  title: 'Message',
  props: {
    width: 20,
  },
  sortByProp: 'message',
  renderExport: (job) => job.message,
};

export const exportableColumns = [SystemColumn, ResultColumn, MessageColumn];

export default [SystemColumn, ResultColumn, MessageColumn];
