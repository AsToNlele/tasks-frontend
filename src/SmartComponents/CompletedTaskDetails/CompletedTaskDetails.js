import React, { useState, useEffect } from 'react';
import { useHistory, useParams } from 'react-router-dom';
import TasksTables from '../../Utilities/hooks/useTableTools/Components/TasksTables';
import { Main } from '@redhat-cloud-services/frontend-components/Main';
import {
  PageHeader,
  PageHeaderTitle,
} from '@redhat-cloud-services/frontend-components/PageHeader';
import {
  Breadcrumb,
  BreadcrumbItem,
  Card,
  Flex,
  FlexItem,
} from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import columns, { exportableColumns } from './Columns';
import { fetchExecutedTask, fetchExecutedTaskJobs } from '../../../api';
import * as Filters from './Filters';
import {
  COMPLETED_INFO_PANEL,
  COMPLETED_INFO_PANEL_FLEX_PROPS,
  COMPLETED_INFO_BUTTONS,
  COMPLETED_INFO_BUTTONS_FLEX_PROPS,
  LOADING_INFO_PANEL,
  LOADING_JOBS_TABLE,
  TASK_ERROR,
  TASKS_TABLE_DEFAULTS,
} from '../../constants';
import FlexibleFlex from '../../PresentationalComponents/FlexibleFlex/FlexibleFlex';
import EmptyStateDisplay from '../../PresentationalComponents/EmptyStateDisplay/EmptyStateDisplay';
import RunTaskModal from '../RunTaskModal/RunTaskModal';
import DeleteCancelTaskModal from '../../PresentationalComponents/DeleteCancelTaskModal/DeleteCancelTaskModal';
import { emptyRows } from '../../PresentationalComponents/NoResultsTable/NoResultsTable';
import { dispatchNotification } from '../../Utilities/Dispatcher';

const CompletedTaskDetails = () => {
  const { id } = useParams();
  const filters = Object.values(Filters);
  const [completedTaskDetails, setCompletedTaskDetails] =
    useState(LOADING_INFO_PANEL);
  const [completedTaskJobs, setCompletedTaskJobs] =
    useState(LOADING_JOBS_TABLE);
  const [error, setError] = useState();
  const [runTaskModalOpened, setRunTaskModalOpened] = useState(false);
  const [selectedSystems, setSelectedSystems] = useState([]);
  const [isDeleteCancelModalOpened, setIsDeleteCancelModalOpened] =
    useState(false);
  const [isDelete, setIsDelete] = useState(false);
  const [isCancel, setIsCancel] = useState(false);
  const history = useHistory();

  const getSelectedSystems = () => {
    return completedTaskJobs.map((job) => job.system);
  };

  const fetchData = async () => {
    let taskDetails = await fetchExecutedTask(id);

    if (isError(taskDetails)) {
      setNotification(taskDetails);
      setError(taskDetails);
    } else {
      const taskJobs = await fetchExecutedTaskJobs(id);

      if (isError(taskJobs)) {
        setNotification(taskJobs);
        setError(taskJobs);
      } else {
        taskDetails.messages_count = taskJobs.data.filter((item) => {
          return item.results.message !== 'No vulnerability found.';
        }).length;
        taskDetails.system_count = taskJobs.data.length;
        await setCompletedTaskDetails(taskDetails);
        await setCompletedTaskJobs(taskJobs.data);
      }
    }
  };

  const isError = (result) => {
    return result?.response?.status && result?.response?.status !== 200;
  };

  const setNotification = (result) => {
    dispatchNotification({
      variant: 'danger',
      title: 'Error',
      description: result.message,
      dismissable: true,
      autoDismiss: false,
    });
  };

  useEffect(() => {
    fetchData();
  }, []);

  useEffect(() => {
    setSelectedSystems(getSelectedSystems());
  }, [completedTaskJobs]);

  useEffect(async () => {
    if (isDelete) {
      history.push('/executed');
      setIsDelete(false);
    }

    if (isCancel) {
      await setCompletedTaskDetails(LOADING_INFO_PANEL);
      await setCompletedTaskJobs(LOADING_JOBS_TABLE);
      await fetchData();
      setIsCancel(false);
    }
  }, [isCancel, isDelete]);

  return (
    <div>
      <RunTaskModal
        description={completedTaskDetails.description}
        error={error}
        isOpen={runTaskModalOpened}
        selectedSystems={selectedSystems}
        setModalOpened={setRunTaskModalOpened}
        slug={completedTaskDetails.task_slug}
        title={completedTaskDetails.task_title}
      />
      <DeleteCancelTaskModal
        id={completedTaskDetails.id}
        isOpen={isDeleteCancelModalOpened}
        setIsCancel={setIsCancel}
        setIsDelete={setIsDelete}
        setModalOpened={setIsDeleteCancelModalOpened}
        startTime={completedTaskDetails.start_time}
        status={completedTaskDetails.status}
        title={completedTaskDetails.task_title}
      />
      {error ? (
        <EmptyStateDisplay
          icon={ExclamationCircleIcon}
          color="#c9190b"
          title={'Task cannot be displayed'}
          text={TASK_ERROR}
          error={`Error ${error?.response?.status}: ${error?.message}`}
        />
      ) : (
        <React.Fragment>
          <PageHeader>
            <Breadcrumb ouiaId="completed-tasks-details-breadcrumb">
              <BreadcrumbItem to="/beta/insights/tasks/executed">
                Tasks
              </BreadcrumbItem>
              <BreadcrumbItem isActive>
                {completedTaskDetails.task_title}
              </BreadcrumbItem>
            </Breadcrumb>
            <Flex direction={{ default: 'column', md: 'row' }}>
              <Flex
                direction={{ default: 'column' }}
                flex={{ default: 'flex_1' }}
              >
                <FlexItem>
                  <PageHeaderTitle title={completedTaskDetails.task_title} />
                </FlexItem>
                <FlexItem>{completedTaskDetails.description}</FlexItem>
              </Flex>
              <FlexibleFlex
                data={completedTaskDetails}
                flexContents={COMPLETED_INFO_BUTTONS(
                  completedTaskDetails.task_slug,
                  setRunTaskModalOpened,
                  //completedTaskDetails.status,
                  setIsDeleteCancelModalOpened
                )}
                flexProps={COMPLETED_INFO_BUTTONS_FLEX_PROPS}
              />
            </Flex>
          </PageHeader>
          <Main>
            <Card>
              <Flex
                className="completed-task-details-info-border"
                justifyContent={{ default: 'justifyContentSpaceBetween' }}
                direction={{ default: 'column', md: 'row' }}
              >
                <FlexibleFlex
                  data={completedTaskDetails}
                  flexContents={COMPLETED_INFO_PANEL}
                  flexProps={COMPLETED_INFO_PANEL_FLEX_PROPS}
                />
              </Flex>
            </Card>
            <br />
            <Card>
              <TasksTables
                label={`${completedTaskDetails.id}-completed-jobs`}
                ouiaId={`${completedTaskDetails.id}-completed-jobs-table`}
                columns={columns}
                items={completedTaskJobs}
                filters={{
                  filterConfig: filters,
                }}
                options={{
                  ...TASKS_TABLE_DEFAULTS,
                  exportable: {
                    ...TASKS_TABLE_DEFAULTS.exportable,
                    columns: exportableColumns,
                  },
                }}
                emptyRows={emptyRows('jobs')}
                isStickyHeader
              />
            </Card>
          </Main>
        </React.Fragment>
      )}
    </div>
  );
};

export default CompletedTaskDetails;
