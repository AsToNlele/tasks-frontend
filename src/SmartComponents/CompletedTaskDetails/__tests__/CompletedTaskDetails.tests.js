import React from 'react';
import {
  act,
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import CompletedTaskDetails from '../CompletedTaskDetails';
import { fetchExecutedTask } from '../../../../api';
import { log4j_task } from './__fixtures__/completedTasksDetails.fixtures';

jest.mock('../../../../api');

describe('CompletedTaskDetails', () => {
  let props;
  let mockStore = configureStore();

  it('should render correctly', async () => {
    const store = mockStore(props);
    fetchExecutedTask.mockImplementation(async () => {
      return log4j_task;
    });

    render(
      <MemoryRouter keyLength={0}>
        <Provider store={store}>
          <CompletedTaskDetails />
        </Provider>
      </MemoryRouter>
    );

    expect(screen.getByText('lalala')).toBeInTheDocument();
  });

  it.skip('should add filter', async () => {
    const store = mockStore(props);
    const { asFragment } = render(
      <MemoryRouter keyLength={0}>
        <Provider store={store}>
          <CompletedTaskDetails />
        </Provider>
      </MemoryRouter>
    );

    const input = screen.getByLabelText('text input');
    fireEvent.change(input, { target: { value: 'A' } });
    expect(input.value).toBe('A');
    await waitFor(() => expect(asFragment()).toMatchSnapshot());
  });

  it.skip('should remove filter', async () => {
    const store = mockStore(props);
    const { asFragment } = render(
      <MemoryRouter keyLength={0}>
        <Provider store={store}>
          <CompletedTaskDetails />
        </Provider>
      </MemoryRouter>
    );

    const input = screen.getByLabelText('text input');
    fireEvent.change(input, { target: { value: 'A' } });
    expect(input.value).toBe('A');
    await waitFor(() => userEvent.click(screen.getByLabelText('close')));
    expect(input.value).toBe('');
    await waitFor(() => expect(asFragment()).toMatchSnapshot());
  });
});
