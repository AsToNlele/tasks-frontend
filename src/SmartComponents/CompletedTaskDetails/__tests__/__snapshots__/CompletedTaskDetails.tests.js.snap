// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedTaskDetails should render expandable rows correctly 1`] = `
<DocumentFragment>
  <div>
    <section
      class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
      widget-type="InsightsPageHeader"
    >
      <nav
        aria-label="Breadcrumb"
        class="pf-c-breadcrumb"
        data-ouia-component-id="completed-tasks-details-breadcrumb"
        data-ouia-component-type="PF4/Breadcrumb"
        data-ouia-safe="true"
      >
        <ol
          class="pf-c-breadcrumb__list"
        >
          <li
            class="pf-c-breadcrumb__item"
          >
            <a
              class="pf-c-breadcrumb__link"
              href="/executed"
            >
              Tasks
            </a>
          </li>
          <li
            class="pf-c-breadcrumb__item"
          >
            <span
              class="pf-c-breadcrumb__item-divider"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </span>
          </li>
        </ol>
      </nav>
      <div
        class="pf-l-flex pf-m-column pf-m-row-on-md"
      >
        <div
          class="pf-l-flex pf-m-flex-1 pf-m-column"
        >
          <div
            class=""
          >
            <h1
              class="pf-c-title pf-m-2xl"
              data-ouia-component-id="OUIA-Generated-Title-1"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
              widget-type="InsightsPageHeaderTitle"
            />
          </div>
          <div
            class=""
          />
        </div>
        <div
          class="pf-l-flex"
        >
          <div
            class=""
          >
            <button
              aria-disabled="false"
              aria-label="undefined-run-task-button"
              class="pf-c-button pf-m-secondary"
              data-ouia-component-id="OUIA-Generated-Button-secondary-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Run task again
            </button>
          </div>
        </div>
        <div
          class="pf-l-flex"
        >
          <div
            class=""
          >
            <div
              class="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-1"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe="true"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Actions"
                class="pf-c-dropdown__toggle pf-m-plain"
                id="executed-task-kebab"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="pf-l-page__main-section pf-c-page__main-section"
      page-type=""
    >
      <article
        class="pf-c-card"
        data-ouia-component-id="OUIA-Generated-Card-1"
        data-ouia-component-type="PF4/Card"
        data-ouia-safe="true"
        id=""
      >
        <div
          class="pf-l-flex pf-m-column pf-m-row-on-md pf-m-justify-content-space-between completed-task-details-info-border"
        >
          <div
            class="pf-l-flex pf-m-column"
          >
            <div
              class=""
            >
              <b>
                Systems
              </b>
            </div>
            <div
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </div>
          </div>
          <div
            class="pf-l-flex pf-m-column"
          >
            <div
              class=""
            >
              <b>
                Run start
              </b>
            </div>
            <div
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </div>
          </div>
          <div
            class="pf-l-flex pf-m-column"
          >
            <div
              class=""
            >
              <b>
                Run end
              </b>
            </div>
            <div
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </div>
          </div>
          <div
            class="pf-l-flex pf-m-column"
          >
            <div
              class=""
            >
              <b>
                Initiated by
              </b>
            </div>
            <div
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </div>
          </div>
          <div
            class="pf-l-flex pf-m-column"
          >
            <div
              class=""
            >
              <b>
                Systems with alerts
              </b>
            </div>
            <div
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </div>
          </div>
        </div>
      </article>
      <br />
      <article
        class="pf-c-card"
        data-ouia-component-id="OUIA-Generated-Card-2"
        data-ouia-component-type="PF4/Card"
        data-ouia-safe="true"
        id=""
      >
        <div
          class="pf-c-toolbar  ins-c-primary-toolbar"
          data-ouia-component-id="PrimaryToolbar"
          data-ouia-component-type="PF4/Toolbar"
          data-ouia-safe="true"
          id="ins-primary-data-toolbar"
        >
          <div
            class="pf-c-toolbar__content"
          >
            <div
              class="pf-c-toolbar__content-section"
            >
              <div
                class="pf-c-toolbar__group pf-m-filter-group ins-c-primary-toolbar__group-filter pf-m-spacer-md pf-m-space-items-lg"
              >
                <div
                  class="pf-c-toolbar__item ins-c-primary-toolbar__filter"
                >
                  <div
                    class="pf-l-split ins-c-conditional-filter"
                  >
                    <div
                      class="pf-l-split__item"
                    >
                      <div
                        class="pf-c-dropdown ins-c-conditional-filter__group"
                        data-ouia-component-id="ConditionalFilter"
                        data-ouia-component-type="PF4/Dropdown"
                        data-ouia-safe="true"
                      >
                        <button
                          aria-expanded="false"
                          aria-haspopup="true"
                          aria-label="Conditional filter"
                          class="pf-c-dropdown__toggle"
                          data-ouia-component-id="ConditionalFilter"
                          data-ouia-component-type="PF4/DropdownToggle"
                          data-ouia-safe="true"
                          id="pf-dropdown-toggle-id-2"
                          type="button"
                        >
                          <span
                            class="pf-c-dropdown__toggle-text"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 512 512"
                              width="1em"
                            >
                              <path
                                d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
                              />
                            </svg>
                            <span
                              class="ins-c-conditional-filter__value-selector"
                            >
                              Status
                            </span>
                          </span>
                          <span
                            class="pf-c-dropdown__toggle-icon"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 320 512"
                              width="1em"
                            >
                              <path
                                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                    </div>
                    <div
                      class="pf-l-split__item pf-m-fill"
                    >
                      <div
                        class="pf-c-select"
                        data-ouia-component-id="Filter by status"
                        data-ouia-component-type="PF4/Select"
                        data-ouia-safe="true"
                      >
                        <button
                          aria-expanded="false"
                          aria-label="Options menu"
                          aria-labelledby=" pf-select-toggle-id-1"
                          class="pf-c-select__toggle"
                          id="pf-select-toggle-id-1"
                          type="button"
                        >
                          <div
                            class="pf-c-select__toggle-wrapper"
                          >
                            <span
                              class="pf-c-select__toggle-text"
                            >
                              Filter by status
                            </span>
                          </div>
                          <span
                            class="pf-c-select__toggle-arrow"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 320 512"
                              width="1em"
                            >
                              <path
                                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="pf-c-toolbar__item pf-m-spacer-sm"
              >
                <div
                  class="pf-c-dropdown"
                  data-ouia-component-id="Export"
                  data-ouia-component-type="PF4/Dropdown"
                  data-ouia-safe="true"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Export"
                    class="pf-c-dropdown__toggle pf-m-plain"
                    data-ouia-component-id="Export"
                    data-ouia-component-type="PF4/DropdownToggle"
                    data-ouia-safe="true"
                    id="pf-dropdown-toggle-id-3"
                    type="button"
                  >
                    <span>
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 1024 1024"
                        width="1em"
                      >
                        <path
                          d="M975.8,636.9 L870.9,741.8 L457.9,328.6 C452.1,322.8 445.4,319.9 437.9,319.9 C430.4,319.9 423.7,322.8 417.9,328.6 L328.8,417.7 C323,423.5 320.1,430.2 320.1,437.7 C320.1,445.2 323,451.9 328.8,457.7 L742,870.7 L636.9,975.8 C610.5,1002.2 619.4,1024 656.8,1024 L956,1024 C1014.5,1024 1024,1013.7 1024,955.9 L1024,656.7 C1023.9,619.4 1002.2,610.5 975.8,636.9 Z M128,128 L896,128 L896,361.7 C896.007942,370.182681 899.389907,378.313788 905.4,384.3 L996.7,475.6 C1006.8,485.7 1024,478.5 1024,464.3 L1024,22.7 C1024,16.1 1021.9,10.7 1017.6,6.4 C1013.3,2.1 1007.9,0 1001.3,0 L22.7,0 C16.1,0 10.7,2.1 6.4,6.4 C2.1,10.7 0,16.1 0,22.7 L0,1001.3 C0,1007.9 2.1,1013.3 6.4,1017.6 C10.7,1021.9 16.1,1024 22.7,1024 L463.4,1024 C469.862884,1023.98894 475.684489,1020.0908 478.156232,1014.11925 C480.627976,1008.14769 479.264428,1001.27548 474.7,996.7 L383.4,905.4 C377.413788,899.389907 369.282681,896.007942 360.8,896 L128,896 L128,128 Z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="pf-c-toolbar__item ins-c-primary-toolbar__pagination"
              >
                <div
                  class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__lg"
                >
                  <span
                    class="pf-u-screen-reader"
                  />
                </div>
              </div>
            </div>
            <div
              class="pf-c-toolbar__expandable-content"
              id="ins-primary-data-toolbar-expandable-content-1"
            >
              <div
                class="pf-c-toolbar__group"
              />
            </div>
          </div>
          <div
            class="pf-c-toolbar__content pf-m-hidden"
            hidden=""
          >
            <div
              class="pf-c-toolbar__group"
            />
          </div>
        </div>
        <table
          aria-label="undefined-completed-jobs"
          class="pf-c-table pf-m-grid-md"
          data-ouia-component-id="undefined-completed-jobs-table"
          data-ouia-component-type="PF4/Table"
          data-ouia-safe="true"
          role="grid"
        >
          <thead
            class=""
          >
            <tr
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-1"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <th
                aria-sort="ascending"
                class="pf-c-table__sort pf-m-selected pf-m-width-30"
                data-key="0"
                data-label="System name"
                scope="col"
              >
                <button
                  class="pf-c-table__button"
                  type="button"
                >
                  <div
                    class="pf-c-table__button-content"
                  >
                    <span
                      class="pf-c-table__text"
                    >
                      System name
                    </span>
                    <span
                      class="pf-c-table__sort-indicator"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z"
                        />
                      </svg>
                    </span>
                  </div>
                </button>
              </th>
              <th
                aria-sort="none"
                class="pf-c-table__sort pf-m-width-10"
                data-key="1"
                data-label="Status"
                scope="col"
              >
                <button
                  class="pf-c-table__button"
                  type="button"
                >
                  <div
                    class="pf-c-table__button-content"
                  >
                    <span
                      class="pf-c-table__text"
                    >
                      Status
                    </span>
                    <span
                      class="pf-c-table__sort-indicator"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                        />
                      </svg>
                    </span>
                  </div>
                </button>
              </th>
              <th
                aria-sort="none"
                class="pf-c-table__sort pf-m-width-35"
                data-key="2"
                data-label="Message"
                scope="col"
              >
                <button
                  class="pf-c-table__button"
                  type="button"
                >
                  <div
                    class="pf-c-table__button-content"
                  >
                    <span
                      class="pf-c-table__text"
                    >
                      Message
                    </span>
                    <span
                      class="pf-c-table__sort-indicator"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                        />
                      </svg>
                    </span>
                  </div>
                </button>
              </th>
            </tr>
          </thead>
          <tbody
            class=""
            role="rowgroup"
          >
            <tr
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-2"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-m-width-30"
                data-key="0"
                data-label="System name"
              >
                <a
                  href="/insights/inventory/undefined"
                >
                  <div
                    class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                  >
                    <span
                      class="pf-u-screen-reader"
                    />
                  </div>
                </a>
              </td>
              <td
                class="pf-m-width-10"
                data-key="1"
                data-label="Status"
              >
                <div
                  class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                >
                  <span
                    class="pf-u-screen-reader"
                  />
                </div>
              </td>
              <td
                class="pf-m-width-35"
                data-key="2"
                data-label="Message"
              >
                <div
                  class="pf-l-split"
                >
                  <div
                    class="pf-l-split__item"
                    color="#A30000"
                  >
                    <div
                      class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                    >
                      <span
                        class="pf-u-screen-reader"
                      />
                    </div>
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-3"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-m-width-30"
                data-key="0"
                data-label="System name"
              >
                <a
                  href="/insights/inventory/undefined"
                >
                  <div
                    class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                  >
                    <span
                      class="pf-u-screen-reader"
                    />
                  </div>
                </a>
              </td>
              <td
                class="pf-m-width-10"
                data-key="1"
                data-label="Status"
              >
                <div
                  class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                >
                  <span
                    class="pf-u-screen-reader"
                  />
                </div>
              </td>
              <td
                class="pf-m-width-35"
                data-key="2"
                data-label="Message"
              >
                <div
                  class="pf-l-split"
                >
                  <div
                    class="pf-l-split__item"
                    color="#A30000"
                  >
                    <div
                      class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                    >
                      <span
                        class="pf-u-screen-reader"
                      />
                    </div>
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-4"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-m-width-30"
                data-key="0"
                data-label="System name"
              >
                <a
                  href="/insights/inventory/undefined"
                >
                  <div
                    class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                  >
                    <span
                      class="pf-u-screen-reader"
                    />
                  </div>
                </a>
              </td>
              <td
                class="pf-m-width-10"
                data-key="1"
                data-label="Status"
              >
                <div
                  class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                >
                  <span
                    class="pf-u-screen-reader"
                  />
                </div>
              </td>
              <td
                class="pf-m-width-35"
                data-key="2"
                data-label="Message"
              >
                <div
                  class="pf-l-split"
                >
                  <div
                    class="pf-l-split__item"
                    color="#A30000"
                  >
                    <div
                      class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                    >
                      <span
                        class="pf-u-screen-reader"
                      />
                    </div>
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-5"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-m-width-30"
                data-key="0"
                data-label="System name"
              >
                <a
                  href="/insights/inventory/undefined"
                >
                  <div
                    class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                  >
                    <span
                      class="pf-u-screen-reader"
                    />
                  </div>
                </a>
              </td>
              <td
                class="pf-m-width-10"
                data-key="1"
                data-label="Status"
              >
                <div
                  class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                >
                  <span
                    class="pf-u-screen-reader"
                  />
                </div>
              </td>
              <td
                class="pf-m-width-35"
                data-key="2"
                data-label="Message"
              >
                <div
                  class="pf-l-split"
                >
                  <div
                    class="pf-l-split__item"
                    color="#A30000"
                  >
                    <div
                      class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                    >
                      <span
                        class="pf-u-screen-reader"
                      />
                    </div>
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-6"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-m-width-30"
                data-key="0"
                data-label="System name"
              >
                <a
                  href="/insights/inventory/undefined"
                >
                  <div
                    class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                  >
                    <span
                      class="pf-u-screen-reader"
                    />
                  </div>
                </a>
              </td>
              <td
                class="pf-m-width-10"
                data-key="1"
                data-label="Status"
              >
                <div
                  class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                >
                  <span
                    class="pf-u-screen-reader"
                  />
                </div>
              </td>
              <td
                class="pf-m-width-35"
                data-key="2"
                data-label="Message"
              >
                <div
                  class="pf-l-split"
                >
                  <div
                    class="pf-l-split__item"
                    color="#A30000"
                  >
                    <div
                      class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                    >
                      <span
                        class="pf-u-screen-reader"
                      />
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </article>
    </section>
  </div>
</DocumentFragment>
`;
